tests/Agent/IntegrationTests/UnboundedIntegrationTests/NServiceBus5/NServiceBus5ReceiveTests.cs

```diff
@@ -0,0 +1,96 @@
+// Copyright 2020 New Relic, Inc. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+
+using NewRelic.Agent.IntegrationTestHelpers;
+using NewRelic.Testing.Assertions;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Xunit;
+using Xunit.Abstractions;
+using Assert = Xunit.Assert;
+using MultiFunctionApplicationHelpers;
+
+namespace NewRelic.Agent.UnboundedIntegrationTests.NServiceBus5
+{
+    public abstract class NServiceBus5ReceiveTestsBase<TFixture> : NewRelicIntegrationTest<TFixture>
+        where TFixture : ConsoleDynamicMethodFixture
+    {
+        private readonly ConsoleDynamicMethodFixture _fixture;
+
+        protected NServiceBus5ReceiveTestsBase(TFixture fixture, ITestOutputHelper output)  : base(fixture)
+        {
+            _fixture = fixture;
+            _fixture.TestLogger = output;
+
+            _fixture.AddCommand("NServiceBusSetup Setup");
+            _fixture.AddCommand("NServiceBusReceiverHost Start");
+            _fixture.AddCommand("NServiceBusService Start");
+            _fixture.AddCommand("NServiceBusService SendValid");
+            _fixture.AddCommand("NServiceBusService SendInvalid");
+            _fixture.AddCommand("RootCommands DelaySeconds 5");
```

Comment:
Hopefully this is reliable. I could see this being a source of issues on slow machines/the cloud

Reply:
I was worried about that too, but I was having trouble with the AgentLog taking forever and needing more than one entry to for the various data types.  I would prefer to extend this style of wait-timer out a bit longer over going back to the AgentLog watcher. 

Thoughts?

Reply:
Let's see how reliable it is, and extend if needed?